packages/eslint-plugin/src/rules/ban-ts-comment.ts

```diff
@@ -71,17 +104,29 @@ export default util.createRule<[Options], MessageIds>({
             return;
           }
 
-          const [, directive] = tsCommentRegExp.exec(comment.value) ?? [];
+          const [, directive, description] =
+            tsCommentRegExp.exec(comment.value) ?? [];
 
           const fullDirective = `ts-${directive}` as keyof Options;
 
-          if (options[fullDirective]) {
+          const option = options[fullDirective];
+          if (option === true) {
             context.report({
               data: { directive },
               node: comment,
               messageId: 'tsDirectiveComment',
             });
           }
+
+          if (option === 'allow-with-description') {
+            if (description.length === 0) {
```

Comment:
it would probably be good to do `description.trim().length > X` where `X` is some threshold.

That way you cannot circumvent the rule with empty space or a single character
```ts
// @ts-ignore        (\n)
// @ts-ignore        .         (\n)
```

Maybe a threshold of like 4 is "good enough" to make sure it's not a completely useless comment?
That way you at least need to put `TODO`.

Reply:
I did this (basically) for https://github.com/typescript-eslint/typescript-eslint/pull/2099/commits/12262fe6c61079569dbf7bf68ea1213cd5d2fdc8

I did 3 so that `// @ts-ignore FIX` would be allowed. 

I'd be happy to add another option:

```
interface Options {
  'ts-expect-error'?: boolean | 'allow-with-description';
  'ts-ignore'?: boolean | 'allow-with-description';
  'ts-nocheck'?: boolean | 'allow-with-description';
  'ts-check'?: boolean | 'allow-with-description';
  threshold: number;
}


const defaultOptions: [Options] = [
  {
    'ts-expect-error': true,
    'ts-ignore': true,
    'ts-nocheck': true,
    'ts-check': false,
    threshold: 3,
  },
];
```

if you think that's best.  I can see how some people might actually _want_ to set it to something like 10 or maybe greater.

Reply:
ok then, I added the option in https://github.com/typescript-eslint/typescript-eslint/pull/2099/commits/65c71a018013841916efad9e8aba4ddcb1352d38